src/main/flight/pid.c

```diff
@@ -614,6 +627,27 @@ void pidAcroTrainerInit(void)
 }
 #endif // USE_ACRO_TRAINER
 
+#ifdef USE_THRUST_LINEARIZATION
+float pidCompensateThrustLinearization(float throttle)
+{
+    if (thrustLinearization != 0.0f) {
+        throttle = throttle * (throttle * thrustLinearization + 1.0f - thrustLinearization);
+    }
+    return throttle;
+}
+
+float pidApplyThrustLinearization(float motorOutput)
+{
+    if (thrustLinearization != 0.0f) {
+        if (motorOutput > 0.0f) {
+            motorOutput = sqrtf(motorOutput * thrustLinearizationReciprocal +
+                                thrustLinearizationB * thrustLinearizationB) - thrustLinearizationB;
+        }
+    }
+    return motorOutput;
+}
+#endif
+
```

Comment:
From a code structure perspective it seems like these functions should be in mixer.c, not pid.c.  They're only called from mixer.c and don't really rely on any PID values. Probably makes sense to move the static variables and initialization into the the mixer code as well.

Reply:
@etracer these functions were actually located in the mixer previously. I moved them to pid.c after deciding that the feature would be better A/B testable if its parameter were in the pidProfile.
Having it in the pid profile requires that the statics get recalculated on pidProfile change. In the end I judged exposing these two functions from the pids.h to be simpler overall.
There's also a second angle: since yaw is mediated via rpm' it does not need to be thrust linearized. Implementing this will likely involve the pids more closely than is now the case. Taking this into account the feature could live in either world.
This actually reminds me of the integratedYaw PR where I initially added it to mixer.c but Boris preferred it in pid.c. Which I finally complied with due to the attractiveness of have the setting in the pidProfile.